shared/cert.go

```diff
@@ -79,19 +85,39 @@ func EncodeCert(c *Cert) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
-func VerifyCertificate(certificate *OpaqueCert, certifierPubKey, nodeID []byte) (*Cert, error) {
-	if !ed25519.Verify(certifierPubKey, certificate.Data, certificate.Signature) {
-		return nil, errors.New("signature mismatch")
-	}
-	decoded, err := certificate.Decode()
-	if err != nil {
-		return nil, fmt.Errorf("decoding: %w", err)
-	}
-	if !bytes.Equal(decoded.Pubkey, nodeID) {
-		return nil, errors.New("pubkey mismatch")
-	}
-	if decoded.Expiration != nil && decoded.Expiration.Before(time.Now()) {
-		return nil, fmt.Errorf("%w at %v", ErrCertExpired, decoded.Expiration)
+func VerifyCertificate(
+	certificate *OpaqueCert,
+	matchingKeys [][]byte,
+	nodeID []byte,
+) (*Cert, error) {
```

Comment:
I think it would be more ergonomic to leave this function as it was (take 1 certificate) and have the user call it many times, for each certificate.